File: src/lib/bindings/video.ts
import { invoke } from "@tauri-apps/api/core";

export const setBrightness = (value: number) => invoke<void>("set_brightness", { value });
export const setContrast = (value: number) => invoke<void>("set_contrast", { value });
export const setSaturation = (value: number) => invoke<void>("set_saturation", { value });
export const toggleDeinterlace = () => invoke<void>("toggle_deinterlace");
export const setVideoZoom = (value: number) => invoke<void>("set_video_zoom", { value });
export const setVideoPan = (x: number, y: number) => invoke<void>("set_video_pan", { x, y });
export const resetVideoZoomPan = () => invoke<void>("reset_video_zoom_pan");

export interface ZoomPanState { zoom: number; pan_x: number; pan_y: number; }
export const getVideoZoomPan = () => invoke<ZoomPanState>("get_video_zoom_pan");

export const setAspectRatio = (ratio: string) => invoke<void>("set_aspect_ratio", { ratio });
export const getAspectRatio = () => invoke<string>("get_aspect_ratio");

File: src-tauri/src/commands/video.rs
use tauri::State;

use crate::error::AppError;
use crate::services::video::{VideoService, ZoomPanState};
use crate::state::MpvState;

#[tauri::command]
pub async fn set_brightness(value: i64, state: State<'_, MpvState>) -> Result<(), AppError> {
    VideoService::set_brightness(state.get()?, value)?;
    Ok(())
}

#[tauri::command]
pub async fn set_contrast(value: i64, state: State<'_, MpvState>) -> Result<(), AppError> {
    VideoService::set_contrast(state.get()?, value)?;
    Ok(())
}

#[tauri::command]
pub async fn set_saturation(value: i64, state: State<'_, MpvState>) -> Result<(), AppError> {
    VideoService::set_saturation(state.get()?, value)?;
    Ok(())
}

#[tauri::command]
pub async fn toggle_deinterlace(state: State<'_, MpvState>) -> Result<(), AppError> {
    VideoService::toggle_deinterlace(state.get()?)?;
    Ok(())
}

#[tauri::command]
pub async fn set_video_zoom(value: f64, state: State<'_, MpvState>) -> Result<(), AppError> {
    VideoService::set_zoom(state.get()?, value)?;
    Ok(())
}

#[tauri::command]
pub async fn set_video_pan(x: f64, y: f64, state: State<'_, MpvState>) -> Result<(), AppError> {
    VideoService::set_pan(state.get()?, x, y)?;
    Ok(())
}

#[tauri::command]
pub async fn get_video_zoom_pan(state: State<'_, MpvState>) -> Result<ZoomPanState, AppError> {
    Ok(VideoService::get_zoom_pan(state.get()?))
}

#[tauri::command]
pub async fn reset_video_zoom_pan(state: State<'_, MpvState>) -> Result<(), AppError> {
    VideoService::reset_zoom_pan(state.get()?)?;
    Ok(())
}

#[tauri::command]
pub async fn set_aspect_ratio(ratio: String, state: State<'_, MpvState>) -> Result<(), AppError> {
    VideoService::set_aspect_ratio(state.get()?, &ratio)?;
    Ok(())
}

#[tauri::command]
pub async fn get_aspect_ratio(state: State<'_, MpvState>) -> Result<String, AppError> {
    Ok(VideoService::get_aspect_ratio(state.get()?))
}
